src/components/bar.js

```diff
@@ -45,31 +47,34 @@ class Bar extends Component {
     lastVisitedAct(currentFolder)
   }
   handleTabulation = () => {
-    const { tabulation, tabular, unsetSelected } = this.props
+    const { tabulation, tabular, setActiveItems } = this.props
     tabulation(!tabular)
-    unsetSelected()
+    setActiveItems([])
   }
 
   handleDelete = () => {
-    const { deleteFolder, activeItems } = this.props
+    const { deleteFolder, activeItems, deleteFile } = this.props
     if (activeItems.length === 1 && activeItems[0].type === ITEM_TYPE.folder) {
       deleteFolder(activeItems[0].obj.id)
     }
+    if (activeItems.length === 1 && activeItems[0].type === ITEM_TYPE.file) {
+      deleteFile(activeItems[0].obj.id)
+    }
     setActiveItems([])
     this.setState({ isDelete: false })
   }
   successCallback = () => {
     this.props.fetchFiles()
```

Comment:
the file fetching should be automatic in the action itself so that we donot have to make callback functions at various components

Reply:
I adopted this successCallback approach because we can also set the state of component to initial state inside it. Try creating 2 folders one after another, there is a need to set the state back to initial.